packages/cli-repl/src/logger.ts

```diff
@@ -31,23 +95,68 @@ function logger(bus: any, logDir: string) {
 
   bus.on('mongosh:use', function(args) {
     log.info(args);
+
+    if (telemetry) {
+      analytics.track({
+        userId,
+        event: 'mongosh:use'
+      });
+    }
   });
 
   bus.on('mongosh:show', function(args) {
     log.info(args);
+
+    if (telemetry) {
+      analytics.track({
+        userId,
+        event: 'mongosh:show'
+      });
+    }
   });
 
   bus.on('mongosh:it', function(args) {
     log.info(args);
+
+    if (telemetry) {
+      analytics.track({
+        userId,
+        event: 'mongosh:it'
+      });
+    }
   });
 
   bus.on('mongosh:setCtx', function(args) {
     log.info(args)
   })
 
-  bus.on('mongosh:api-call', function(args) {
+  bus.on('mongosh:api-call', function(args: ApiCall) {
     log.info(redactInfo(args));
+
+    // analytics properties to include if they are present in an api-call
+    let properties: ApiCall = {};
+    properties.arguments = {};
+    if (args.method) properties.method = args.method;
+    if (args.class) properties.class = args.class;
+    if (args.arguments.pipeline) properties.arguments.pipeline = args.arguments.pipeline;
+    if (args.arguments.query) properties.arguments.query = args.arguments.query;
+    if (args.arguments.options) properties.arguments.options = args.arguments.options;
+    if (args.arguments.filter) properties.arguments.filter= args.arguments.filter;
+
+    if (telemetry) {
+      analytics.track({
+        userId,
+        event: 'mongosh:api-call',
+        properties: redactInfo(properties)
```

Comment:
@mmarcon so these are the telemetry tracking events for api-call.

`properties` would include for example:

```js
{
  method: 'aggregation',
  class: 'Collection',
  arguments: {
    pipeline: [ {$match: {"lat": {$gt: 45}}} ]
    options: { writeConcern: 1 }
  }
```

I am always omitting `db` and `coll` from analytics. 

Depending on the method, arguments could `filter` or `query` instead of `pipeline`. Or there won't be any, if a method doesn't support any. 

This will send all the api-calls to telemetry. If you want, we can be more specific, and send only specific ones over.

Reply:
@lrlna are we redacting stuff? E.g. if a pipeline or a filter contains email addresses, we should not see them in analytics.

Reply:
Oh yeah, looks like properties are being redacted. Cool.

Reply:
yep! let me know if something that should be redacted, isn't. Can patch up `mongodb-redacted`